### PR TITLE
Add Interface to archiveRoom/unarchiveRoom functions

### PR DESCRIPTION
**Pull request by matthias-brun**
This PR adds an Archive/Unarchive button to the room settings and makes archived rooms only show up in the history. I also slightly changed the layout of the history in order to separate archived from non-archived channels.
screen shot 2015-12-10 at 13 07 23
screen shot 2015-12-10 at 13 08 09

### CODE REVIEW

**Comment by graywolf336** [comment]
Do archived channels display when you click on the "more channels"?

**Comment by matthias-brun** [comment]
No, archived channels are only displayed in the history. I feel like this is the main advantage archiving offers because it allows to show only relevant channels. Channels which are project-specific for example and need to be kept for the record but aren't actively used anymore can be archived so they don't clutter the overview of all accessible channels.

**Comment by graywolf336** [comment]
Okay cool. What happens when someone wants to create a new channel and that new channel has the same name as an archived one? In my mind the expected result would be the new channel is allowed to be created, but that's just my opinion...

**Comment by matthias-brun** [comment]
I think allowing to create new channels with the same name as an archived one might lead to confusion, especially considering that there would be no way to differentiate the two if the new channel were to be archived as well. So doing that is currently not possible.

**Comment by graywolf336** [comment]
Before I continue, thank you for submitting this pull request. :+1: I'm just picking at things since I haven't had time to actually test it. If someone tries currently to create a channel with the same name as an archived one, with this pull request, what will happen? Will they be given a message saying "Hey a channel by that name exists already but it is archived."? Or will it just say "Another channel by that name exists."? And how does one unarchive a room?

**Comment by matthias-brun** [comment]
No bother, I appreciate hearing someone else's opinion on it. Unarchiving works simply by replacing the "Archive" button with an "Unarchive" button in archived channels.
screen shot 2015-12-11 at 07 17 39
As for creating duplicate channels, that's a good point. I didn't consider that seeing an error message without the channel being in the channel list might be confusing as well. I just had a look at the code and I think it would be a relatively simple change. I'll have a look at that either later today or over the weekend.

**Comment by matthias-brun** [comment]
There is now a custom error message informing the user that the room name is already occupied by an archived one.
screen shot 2015-12-11 at 17 40 26
screen shot 2015-12-11 at 17 40 43
Thanks @graywolf336 for the suggestion!

**Comment by graywolf336** [comment]
@matthias-brun I will be trying to review this item this weekend and test it out. My only other thought was: Is there a message saying "{username} achieved this channel on {date}"?

**Comment by matthias-brun** [comment]
@graywolf336 I thought of adding such a message initially but forgot to actually make the change. There is currently no such message but I'll make sure to add one as soon as I get around to it. I am also going to push a change to give the admin panel for room management the same treatment the users room history received. I assumed that they share code but forgot to check. Apparently they do not.
If there is anything else, let me know. Appreciate the feedback.

**Comment by marceloschmidt** [comment]
I have made a few changes to the top portion of the room settings (now Room Info) layout. I can solve the conflicts once we merge. Would you like me to test and merge this PR or are you still going to do something to it?

**Comment by matthias-brun** [comment]
@marceloschmidt Just the two changes I mentioned above. But if you feel like this is good enough to be merged, by all means go ahead. I can still open a new pull request later on, which will also be easier to merge as it introduces smaller changes rather than changing all at once.
tldr: Sure, feel free!

**Comment by marceloschmidt** [comment]
@matthias-brun merged it, please create a new PR for your changes later on ;) Thanks!!
